server/channels/app/post.go

```diff
@@ -515,6 +515,28 @@ func (a *App) FillInPostProps(rctx request.CTX, post *model.Post, channel *model
 		post.AddProp(model.PostPropsGroupHighlightDisabled, true)
 	}
 
+	// Populate AI-generated username from provided user ID
+	if aiGenUserID, ok := post.GetProp(model.PostPropsAIGeneratedByUserID).(string); ok && aiGenUserID != "" {
+		user, err := a.GetUser(aiGenUserID)
+		if err != nil {
+			// If user doesn't exist, remove the ai_generated_by prop to avoid storing invalid data
+			rctx.Logger().Warn("Failed to get user for AI-generated post, removing ai_generated_by prop", mlog.String("user_id", aiGenUserID), mlog.Err(err))
+			post.DelProp(model.PostPropsAIGeneratedByUserID)
+		} else {
+			// Only allow AI-generated username if the user is the post creator or a bot
+			if user.Id == post.UserId || user.IsBot {
+				post.AddProp(model.PostPropsAIGeneratedByUsername, user.Username)
+			} else {
+				// User ID cannot be a different non-bot user
+				rctx.Logger().Warn("AI-generated user ID must be post creator or bot, removing ai_generated_by prop",
```

Comment:
It looks like can hard fail here by returning an error. Maybe this is worth a hard fail since this should never happen?

Reply:
Yeah that makes sense